exp/services/ledgerexporter/internal/ledger_meta_archive.go

```
@@ -0,0 +1,84 @@
+	// file name
+	objectKey string
+	// Actual binary data
+	data xdr.LedgerCloseMetaBatch
+}
+
+// NewLedgerMetaArchive creates a new LedgerMetaArchive instance.
+func NewLedgerMetaArchive(key string, startSeq uint32, endSeq uint32) *LedgerMetaArchive {
+	return &LedgerMetaArchive{
+		objectKey: key,
+		data: xdr.LedgerCloseMetaBatch{
+			StartSequence: xdr.Uint32(startSeq),
+			EndSequence:   xdr.Uint32(endSeq),
+		},
+	}
+}
+
+// AddLedger adds a LedgerCloseMeta to the archive.
+func (f *LedgerMetaArchive) AddLedger(ledgerCloseMeta xdr.LedgerCloseMeta) error {
+	if ledgerCloseMeta.LedgerSequence() < uint32(f.data.StartSequence) ||
+		ledgerCloseMeta.LedgerSequence() > uint32(f.data.EndSequence) {
+		return fmt.Errorf("ledger sequence %d is outside valid range [%d, %d]",
+			ledgerCloseMeta.LedgerSequence(), f.data.StartSequence, f.data.EndSequence)
+	}
+
+	if len(f.data.LedgerCloseMetas) > 0 {
+		lastSequence, _ := f.GetPreviousLedgerSequence()
+		if ledgerCloseMeta.LedgerSequence() != lastSequence+1 {
+			return fmt.Errorf("ledgers must be added sequentially: expected sequence %d, got %d",
+				lastSequence+1, ledgerCloseMeta.LedgerSequence())
+		}
+	}
+	f.data.LedgerCloseMetas = append(f.data.LedgerCloseMetas, ledgerCloseMeta)
+	return nil
+}
+
+// GetLedgerCount returns the number of ledgers currently in the archive.
+func (f *LedgerMetaArchive) GetLedgerCount() uint32 {
+	return uint32(len(f.data.LedgerCloseMetas))
+}
+
+// GetStartLedgerSequence returns the starting ledger sequence of the archive.
+func (f *LedgerMetaArchive) GetStartLedgerSequence() uint32 {
+	return uint32(f.data.StartSequence)
+}
+
+// GetEndLedgerSequence returns the ending ledger sequence of the archive.
+func (f *LedgerMetaArchive) GetEndLedgerSequence() uint32 {
+	return uint32(f.data.EndSequence)
+}
+
+// GetPreviousLedgerSequence returns the sequence of the last ledger in the archive.
+func (f *LedgerMetaArchive) GetPreviousLedgerSequence() (uint32, error) {
```

this function is only used in one place,`AddLedger()`, and during that invocation the return error is ignored. So I think it would be cleaner to just inline `f.data.LedgerCloseMetas[len(f.data.LedgerCloseMetas)-1].LedgerSequence()` into the `AddLedger()` function

---

Addressed in 708e7d1e2f51e96f5700022de1de67fa69952422